crd-convert/README.md

```diff
@@ -0,0 +1,79 @@
+# CRD conversion
+
+With Strimzi's CRD evolving, there needs to be a way to easily migrate existing CR configuration to the new CRD schema.
+For this purpose a custom Converter for each CRD needs to be implemented.
+
+There are two methods to implement:
+
+```
+public void convertTo(T instance, ApiVersion toVersion)
+public void convertTo(JsonNode node, ApiVersion toVersion)
+```
+
+The first one does conversion directly on the CR instance, which might later lead to reordered configuration output (think "git merge" problems).
+That's why we added the second method, doing the conversion directly on the CR' JsonNode instance, preserving the configuration output as much as possible. 
+
+The code already provides useful conversion methods, such as `move`, `delete`, `replace`. Hence you mostly just need to implement part of `replace` method; e.g. the actual replacement value.
+
+Current Converters are:
+* [KafkaConverter](src/main/java/io/strimzi/kafka/crd/convert/converter/KafkaConverter.java)
+
+### CRD conversion tooling / CLI
+
+To simplify migration as much as possible, we added a new conversion CLI tool:
+
+```
+% crd --help                                                                                                                     
+Usage: crd [-hV] [COMMAND]
+Simple entry command
+  -h, --help      Show this help message and exit.
+  -V, --version   Print version information and exit.
+Commands:
+  help        Displays help information about the specified command
+  convert, c  Convert CRDs
+  
+% crd c     
+Missing required option: '--to-version=<toApiVersion>'
+Usage: crd convert [-d] [-uo] [-fv=<fromApiVersion>] [-it=<inputType>]
+                   [-ll=<level>] [-n=<namespace>] [-o=<outputFile>]
+                   [-ot=<outputType>] -tv=<toApiVersion> [-f=<inputFile> |
+                   -c=<content>] [<Kube API params>...]
+Convert CRDs
+      [<Kube API params>...]
+
+  -c, --content=<content>   CRD content
+  -d, --debug               Use debug?
+  -f, --file=<inputFile>    CRD input file
+      -fv, --from-version=<fromApiVersion>
+                            From K8s ApiVersion - required when used with Kube
+                              API
+      -it, --input-type=<inputType>
+                            Content input type, e.g. json or yaml
+      -ll, --log-level=<level>
+                            Set log level to enable logging
```

Comment:
Why are these indented?

Reply:
This was direct Picocli output ... let me indent it ...